### ISSUE TITLE
Enable x86 on x64

### ISSUE THREAD

**Issue by robotdad** [Microsoft/VSLinux#25]

**Comment by stanthomas** [comment]
I'd like to propose a simple relaxation of matching the build target with remote system architecture. It seems this would be quick to do and addresses the needs of most users.

# Background
Linux and Windows are quite happy running 32-bit applications on 64-bit systems. Indeed, there are good arguments that applications that can comfortably run
in a 2GB address space should be always be built in 32-bit because they are smaller and faster. It's worth noting that Visual Studio itself is a 32-bit application.

By default, GCC will build as 64-bit on 64-bit systems and as 32-bit on 32-bit systems. But it is simple to build as 32-bit on 64-bit systems : just add the g++ command line option -m32.

Note that this is quite distinct from cross-compilation, where binaries can be generated for a different target architecture. GCC can do this too, but that's issue #19.


Obviously a 64-bit Linux system must also have the 32-bit development headers/libraries installed
otherwise compiler, linker and run-time errors will ensue. I suspect this is the reason VCLinux is so rigid about matching project settings to the Linux target native architecture.

# Proposed Change
I suggest that, when a 64-bit target is detected, VCLinux allows building of both 64-bit and 32-bit projects. With the proviso that specifying the compiler option (-m32) and ensuring the presence of 32-bit development headers/libraries is the sole responsibility of the user. Although, in the longer term, this could be something for consideration under Issue #14.
This could be emphasised by modifying the build output message so that:
```
Validating architecture
   Current project architecture 'x64' is compatible with the remote machine architecture 'x64'.
```
would become:
```
Validating architecture
   Current project architecture 'x32', remote machine architecture 'x64' - ensure remote is configured for 32-bit builds.
```

**Comment by KN4CK3R** [comment]
Quickfix if you don't want to wait until the official support.
Remove the check in C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Application Type\Linux\1.0\Linux.Common.targets:
```
  <Target Name="_ValidateValidArchitecture">
    <!--<ValidateValidArchitecture
      Platform="$(Platform)"
      RemoteTarget="$(RemoteTarget)" />-->
  </Target>
```

**Comment by TheLastRar** [comment]
The only project I use this on is x86 (outside constraint), I would rather be able to build and debug on an x64 machine without having to use the workaround posted above

**Comment by stanthomas** [comment]
@lukka
I do not agree with your perfunctory closing of this issue. @robotdad placed this on your *todo* list and you should report on that first. Either 'adressed in release xx.xx by *change detail*' **or** 'will not be actioned because *reason*'. The passage of time is irrelevant; it is quite possible that others have found this topic and concluded that a resolution is on its way.

**Comment by vmcall** [comment]
This has been an issue for over four years now, the above workaround still works but god forbid fixing this issue.

**Comment by lukka** [comment]
@stanthomas @vmcall thanks for the feedback. My suggestion is to report the problem on the [Developer Community forum](https://developercommunity.visualstudio.com/spaces/8/index.html), just copying and pasting what you wrote in here should be enough, and up vote it. 
I understand that you wonder why you need to report again what you already reported: reason is that we are driven by the Dev Community's  ticket, and since this particular issue has never surfaced in the forum, low are the chances that we take action upon this specific problem.

**Comment by Anton-V-K** [comment]
Has anyone already reported this issue to Dev Community Forum?
It is still not solved with VS2017 (15.9.23). Probably it is fixed for VS2019?
The error message as it is shown by MSBuild:

> C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Application Type\Linux\1.0\Linux.targets(151,5): error : Current project architecture 'x86' is incompatible with the remote system architecture 'x64' ('x64'). Please switch the project architecture to 'x64' in Configuration Manager.
